src/XRP/default-xrp-client.ts

```diff
@@ -456,6 +458,43 @@ export default class DefaultXrpClient implements XrpClientDecorator {
     return await this.getTransactionResult(transactionHash)
   }
 
+  /**
+   * Enables DepositPreauth and authorizes an XRPL account to deposit to this XRPL account.
+   *
+   * @set https://xrpl.org/depositpreauth.html
+   *
+   * @param senderXAddress The X-Address of the sender to enable DepositPreauth for.
+   * @param wallet The wallet associated with the XRPL account enabling DepositPreauth and that will sign the request.
+   */
+  public async authorizeDepositPreauth(
+    senderXAddress: string,
```

Comment:
i renamed this to `senderXAddress` from xAddressToAuthorize` to match with the docs but it's way more confusing and i regret this choice. i will change it back